python/dgl/graphbolt/minibatch.py

```diff
@@ -459,98 +459,120 @@ def _to_dgl_blocks(self):
-                    )[0].size(1)
-                    minibatch.negative_node_pairs = {
-                        etype: (
-                            neg_src.view(-1),
-                            self.compacted_node_pairs[etype][
-                                1
-                            ].repeat_interleave(negative_ratio),
-                        )
-                        for etype, neg_src in self.compacted_negative_srcs.items()
-                    }
-            elif self.compacted_negative_dsts is not None:
-                # For homogeneous graph.
-                if isinstance(self.compacted_negative_dsts, torch.Tensor):
-                    negative_ratio = self.compacted_negative_dsts.size(1)
-                    minibatch.negative_node_pairs = (
-                        self.compacted_node_pairs[0].repeat_interleave(
+                    for etype, neg_src in self.compacted_negative_srcs.items()
+                }
+        elif (
+            self.compacted_negative_dsts is not None
+            and self.compacted_node_pairs is not None
+        ):
+            # For homogeneous graph.
+            if isinstance(self.compacted_negative_dsts, torch.Tensor):
+                negative_ratio = self.compacted_negative_dsts.size(1)
+                negative_node_pairs = (
+                    self.compacted_node_pairs[0].repeat_interleave(
+                        negative_ratio
+                    ),
+                    self.compacted_negative_dsts.view(-1),
+                )
+            # For heterogeneous graph.
+            else:
+                negative_ratio = list(self.compacted_negative_dsts.values())[
+                    0
+                ].size(1)
+                negative_node_pairs = {
+                    etype: (
+                        self.compacted_node_pairs[etype][0].repeat_interleave(
                             negative_ratio
                         ),
-                        self.compacted_negative_dsts.view(-1),
+                        neg_dst.view(-1),
                     )
-                # For heterogeneous graph.
-                else:
-                    negative_ratio = list(
-                        self.compacted_negative_dsts.values()
-                    )[0].size(1)
-                    minibatch.negative_node_pairs = {
-                        etype: (
-                            self.compacted_node_pairs[etype][
-                                0
-                            ].repeat_interleave(negative_ratio),
-                            neg_dst.view(-1),
-                        )
-                        for etype, neg_dst in self.compacted_negative_dsts.items()
-                    }
-        return minibatch
+                    for etype, neg_dst in self.compacted_negative_dsts.items()
+                }
+        else:
+            negative_node_pairs = None
+        return negative_node_pairs
+
+    def get_training_node_pair_and_labels(self):
```

Comment:
If node_pair and labels exist, we should directly return those 2 instead of converting from positive and negative node pairs.